opm/autodiff/NonlinearSolver_impl.hpp

```diff
@@ -260,24 +260,28 @@ namespace Opm
         dxOld = dx;
 
         switch (relaxType()) {
-            case DAMPEN:
+            case DAMPEN: {
                 if (omega == 1.) {
                     return;
                 }
-                for (size_t i = 0; i < dx.size(); ++i) {
+                auto i = dx.size();
+                for (i = 0; i < dx.size(); ++i) {
                     dx[i] *= omega;
                 }
                 return;
-            case SOR:
+            }
+            case SOR: {
                 if (omega == 1.) {
                     return;
                 }
-                for (size_t i = 0; i < dx.size(); ++i) {
+                auto i = dx.size();
+                for (i = 0; i < dx.size(); ++i) {
```

Comment:
in @atgeirr's and my opinion this is more correct, but slightly more verbose for no real benefit, but I don't really mind (and @atgeirr probably not either).

BTW: do you happen to now if this works if `size()` returns `const int`?